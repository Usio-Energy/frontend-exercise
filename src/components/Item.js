import React, { Component } from 'react';

export default class Item extends Component {
  constructor() {
    super();

    this.onRemove = this.onRemove.bind(this);
    this.onStatusChange = this.onStatusChange.bind(this);
  }

  onRemove() {
    this.props.onRemove(this.props.id);
  }

  onStatusChange({ target: { checked } }) {
    this.props.onStatusChange(this.props.id, checked);
  }

  render() {
    return (
      <li>
        <input type="checkbox" onChange={ this.onStatusChange }/> {this.props.value}
        &nbsp; - &nbsp;
        <button id="remove-btn" onClick={ this.onRemove }>Remove</button>
      </li>
    );
  }
}