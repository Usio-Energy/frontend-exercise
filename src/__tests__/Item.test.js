import React from 'react';
import { shallow } from 'enzyme';

import Item from '../components/Item';

describe('Item', () => {
  let component;
  let itemId = 45;

  beforeAll(() => {
    component = shallow(
      <Item
        value="Task label"
        isComplete={ false }
        id={ itemId }
        onRemove={ jest.fn() }
        onStatusChange={ jest.fn() }
      />
    );
  });

  it('should render correctly', () => {
    expect(component.getElement()).toMatchSnapshot();
  });

  it('should call `onRemove` prop and pass the item\'s ID when `Remove` button is clicked', () => {
    const removeBtn = component.find('#remove-btn');

    removeBtn.simulate('click');

    expect(component.instance().props.onRemove).toHaveBeenCalledWith(itemId);
  });

  it('should call `onStatusChange` prop and pass the item\'s ID when the checkbox is checked/unchecked', () => {
    const checkbox = component.find('input');

    checkbox.simulate('change', {
      target: {
        checked: true
      }
    });

    expect(component.instance().props.onStatusChange).toHaveBeenCalledWith(itemId, true);
  });
});
