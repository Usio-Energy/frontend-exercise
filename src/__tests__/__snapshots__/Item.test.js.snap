// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<li>
  <input
    onChange={[Function]}
    type="checkbox"
  />
   
  Task label
    -  
  <button
    id="remove-btn"
    onClick={[Function]}
  >
    Remove
  </button>
</li>
`;
